File: core/src/main/java/com/lomekwi/cave/util/FileNameUtil.java
package com.lomekwi.cave.util;

import java.io.File;

public final class FileNameUtil {
    private FileNameUtil(){}
    public static String getExtension(File file) {
        if (file == null) return "";
        String name = file.getName();
        int dotIndex = name.lastIndexOf('.');
        return (dotIndex >= 0 && dotIndex < name.length() - 1) ? name.substring(dotIndex + 1) : "";
    }
    public static String ensureExtension(String name, String ext) {
        if (name == null || ext == null) return name;
        String dotExt = ext.startsWith(".") ? ext : "." + ext;
        return name.toLowerCase().endsWith(dotExt.toLowerCase()) ? name : name + dotExt;
    }
    public static File ensureExtension(File file, String ext) {
        if (file == null) return null;
        String name = file.getName();
        String newName = ensureExtension(name, ext);
        return name.equals(newName) ? file : new File(file.getParent(), newName);
    }
}

File: core/src/main/scala/com/lomekwi/cave/util/FileNameUtil.scala
package com.lomekwi.cave.util

import java.io.File

object FileNameUtil {
  def getExtension(file: File): String = {
    if (file == null) {
      ""
    } else {
      val name = file.getName
      val dotIndex = name.lastIndexOf('.')
      if (dotIndex >= 0 && dotIndex < name.length() - 1) name.substring(dotIndex + 1) else ""
    }
  }
  def ensureExtension(name: String, ext: String): String = {
    if (name == null || ext == null) {
      name
    } else {
      val dotExt = if (ext.startsWith(".")) ext else "." + ext
      if (name.toLowerCase().endsWith(dotExt.toLowerCase())) name else name + dotExt
    }
  }
  def ensureExtension(file: File, ext: String): File = {
    if (file == null) {
      null
    } else {
      val name = file.getName
      val newName = ensureExtension(name, ext)
      if (name.equals(newName)) file else new File(file.getParent, newName)
    }
  }
}
